wallet/src/de/schildbach/wallet/livedata/EncryptWalletLiveData.kt

```diff
@@ -95,10 +95,6 @@ class EncryptWalletLiveData(application: Application) : MutableLiveData<Resource
                 // initialize the authentication key chains to allow recovery of the username
                 wallet.initializeAuthenticationKeyChains(decryptedSeed, newKey);
 
-                if(initialize) {
-                    walletApplication.saveWalletAndFinalizeInitialization()
-                }
```

Comment:
This was moved out of the background thread therefore we should retest all the wallet initialization scenarios.